### PR TITLE
feat: Add forwards

### PR DESCRIPTION
**Pull request by Rushaway**
use case exemple: https://github.com/Rushaway/ZLeader-Remake/blob/3.4/addons/sourcemod/scripting/ZLeader.sp#L560

### CODE REVIEW

File: addons/sourcemod/scripting/include/Spectate.inc [hunk 1]
```diff
@@ -8,6 +8,21 @@
...
 
+/**
+ * Called right before a client was force suicide before moving to spec
+ */
+forward void Spectate_OnCommitSuicide(int client);
+
+/**
+ * Called right after a client was moving to spec
+ */
+forward void Spectate_OnClientMovedToSpec(int client);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/include/Spectate.inc]: you can hook player_team with HookEvent to know this:
- https://sourcemod.dev/#/events/function.HookEvent
- https://wiki.alliedmods.net/Generic_Source_Events#player_team

File: addons/sourcemod/scripting/include/Spectate.inc [hunk 2]
```diff
@@ -8,6 +8,21 @@
...
+
+/**
+ * Called right after a client was switched back to a team
+ */
+forward void Spectate_OnClientSwitchedBack(int client);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/include/Spectate.inc]: you can hook player_team with HookEvent to know this:
- https://sourcemod.dev/#/events/function.HookEvent
- https://wiki.alliedmods.net/Generic_Source_Events#player_team

File: addons/sourcemod/scripting/include/Spectate.inc [hunk 3]
```diff
@@ -8,6 +8,21 @@
 #endif
 #define _Spectate_included
 
+/**
+ * Called right before a client was force suicide before moving to spec
+ */
+forward void Spectate_OnCommitSuicide(int client);
```
Comment (maximee-leroyy) [inline addons/sourcemod/scripting/include/Spectate.inc:14]: you can hook player_death with HookEvent to know this:
- https://sourcemod.dev/#/events/function.HookEvent
- https://wiki.alliedmods.net/Generic_Source_Events#player_death

Comment (Rushaway) [reply addons/sourcemod/scripting/include/Spectate.inc:14]: It's called before, not after, that why I need a forward and not hooking the player death. I need this for this case:
https://github.com/Rushaway/ZLeader-Remake/blob/3.4/addons/sourcemod/scripting/ZLeader.sp#L551C1-L558C2

Comment (maximee-leroyy) [reply addons/sourcemod/scripting/include/Spectate.inc:14]: so you want to let the player lead if he goes to spec?

You can hook `player_team` and check after the player commits suicide if he has gone to spectator (with post event  hook or timer)

**Pull request closed without merging**